notification_controller/controller/controller.go

```diff
@@ -77,6 +79,9 @@ func NewController(
 			if !ok {
 				return false, ""
 			}
+			if namespace != app.GetNamespace() && !glob.MatchStringInList(applicationNamespaces, app.GetNamespace(), false) {
+				return true, "app is not in one of the application-namespaces, nor the notification controller namespace"
+			}
```

Comment:
Can we get test coverage on this? If a full test of the `notificationController` would be prohibitively complex, we could pull out this function and just unit-test that.